listener/listener.go

```diff
@@ -224,8 +233,12 @@ func (l *Listener) Listen(ctx context.Context, handler message.Handler, topicNam
 		_ = topic.Close(ctx)
 	}()
 
+	if l.concurrency == 0 {
+		l.concurrency = 1 //RetryLater won't work with this. Come back and change after talking to strebec?
+	}
+	prefetch := servicebus.SubscriptionWithPrefetchCount(uint32(l.concurrency))
```

Comment:
I don't think prefetch is strictly like to concurrency.
prefetch will make the request fetch N messages at once. that's much better obviously.
but since the messages can be processed asynchronously from the fetching, it's not a requirement.

- fetch message1
- start process message 1
- fetch message2
- start process message 2
- done process message 2
- done process message 1

the above is concurrent processing with a prefetch of 1.

- fetch message1 and message2
- start process message 1
- start process message 2
- done process message 2
- done process message 1

The above is concurrent processing with prefetch of 2 (assuming the messages are present in the subscription when the receiver starts)

Reply:
Agree it's not strictly tied though probably a good default.
Not sure this PR is useful to autoupgrader or replacing the rp without lock auto-renewal.   Take a look at message pump here to see concurrency and auto renewal. I kind of want to get that up to the servicebus sdk as an optional handler any consumer can use. 

https://github.com/Azure/go-shuttle/pull/13/files#diff-f8191b0240dc42b4c68b583cf4f8c33a1c63d8bd3537410acdb0f142a25ec939